students/mattmaeda/session08/circle.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+"""
+Circle example
+"""
+from math import pi
+
+class Circle(object):
+    """ Circle class
+    """
+    _radius = None
+
+    def __init__(self, radius):
+        """
+        :param radius: radius of circle
+        :type: int
+        """
+        self.radius = radius
+
+    @classmethod
+    def from_diameter(cls, diameter):
+        """
+        :param diameter: the circle's diameter
+        :type: int or float
+        """
+        if Circle.is_valid_value(diameter):
+            return Circle(diameter/2)
+        raise ValueError("invalid diameter")
+
+
+    @staticmethod
+    def is_valid_value(value):
```

Comment:
you don't usually do a lot of checking for type in python.

If you want to assure that you have a proper value for, example, the radius, do a:

radius = float(radius)

and it'll automagically give you a ValueError.

And if you REALLY need to know it's a number, than you can do:

isinstance(x, Number)

See:

https://docs.python.org/3/library/numbers.html#module-numbers

But again, I'd mostly just stick with duck typing -- let people set radius to anything they want, and it'll fail when they try to use it.